apps/cowswap-frontend/src/modules/fiatAmount/hooks/useTradeFiatAmounts.ts

```diff
@@ -0,0 +1,21 @@
+import { useIsWrapOrUnwrap } from 'modules/trade/hooks/useIsWrapOrUnwrap'
+
+import { TradeAmounts } from 'common/types'
+import { isFractionFalsy } from 'utils/isFractionFalsy'
+
+import { FiatAmountInfo, useFiatAmount } from './useFiatAmount'
+
+export interface TradeUSDAmounts {
+  inputAmount: FiatAmountInfo
+  outputAmount: FiatAmountInfo
+}
+
+export function useTradeFiatAmounts({ inputAmount, outputAmount }: Partial<TradeAmounts>): TradeUSDAmounts {
+  const isWrapOrUnwrap = useIsWrapOrUnwrap()
+  const isTradeReady = !isWrapOrUnwrap && !isFractionFalsy(inputAmount) && !isFractionFalsy(outputAmount)
```

Comment:
The goal of this hook is to prevent loading fiat amounts when a trade is not ready.
It doesn't make sense to display fiat amount of sell token when there is no but token.

Reply:
This is how it works now on prod ¯\_(ツ)_/¯
But on prod we only don't display values but still do requests